### ISSUE TITLE
Make Terraform generate and deploy layers from S3

### ISSUE THREAD

**Issue by danmermel** [glynnbird/choirless#28]
TBC exactly how...

**Comment by glynnbird** [comment]
I've made Terraform deploy layers from zips on S3. This is great, but the question remains how do those zips _get to S3 in the first place_.

At the moment

- user runs `./lambda/pipeline/build_ffmpeg_layer.sh` to create ffmpeg.zip & ffprobe.zip - the ffmpeg binaries layers. This script is clever enough to check the latest John Van Sickle ffmpeg version, so that it doesn't do necessary work.
- user runs `./lambda/api/build_layer.sh` to create nodejs.zip - the nodejs dependencies layer. This script does and `npm install` and zips up node_modules. 
- terraform apply copies the zips to S3 and the layers are deployed from S3

I tried getting Terraform to call the shell scripts like so:

```
resource "null_resource" "runJVS" {
  provisioner "local-exec" {
    command = "cd ../lambda/pipeline; ./build_ffmpeg_layer.sh"
  }
}
```

it works, but it doesn't run the script on every `terraform apply`. I suspect we'll have to modify our `apply.sh` script to call the two shell scripts before running `terraform apply`
